cmd/copy.go

```diff
@@ -816,180 +813,27 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 		CredentialInfo: cca.credentialInfo,
 	}
 
-	// TODO remove this copy pasted code during refactoring
 	from := cca.fromTo.From()
 	to := cca.fromTo.To()
+
 	// Strip the SAS from the source and destination whenever there is SAS exists in URL.
 	// Note: SAS could exists in source of S2S copy, even if the credential type is OAuth for destination.
-	switch from {
-	case common.ELocation.Local():
-		tmpSrc, err := filepath.Abs(cca.source)
-		if err != nil {
-			return fmt.Errorf("couldn't get absolute path of the source location %s. Failed with errror %s", cca.source, err.Error())
-		}
-
-		tmpSrc = common.ToExtendedPath(tmpSrc)
-
-		jobPartOrder.SourceRoot = cleanLocalPath(getPathBeforeFirstWildcard(tmpSrc))
-		cca.source = cleanLocalPath(tmpSrc)
-
-	case common.ELocation.Benchmark():
-		// noop
-
-	case common.ELocation.Blob():
-		fromUrl, err := url.Parse(cca.source)
-		if err != nil {
-			return fmt.Errorf("error parsing the source url %s. Failed with error %s", fromUrl.String(), err.Error())
-		}
-		blobParts := azblob.NewBlobURLParts(*fromUrl)
-		cca.sourceSAS = blobParts.SAS.Encode()
-		jobPartOrder.SourceSAS = cca.sourceSAS
-		blobParts.SAS = azblob.SASQueryParameters{}
-		bUrl := blobParts.URL()
-		cca.source = bUrl.String()
-
-		// set the clean source root
-		if strings.Contains(blobParts.ContainerName, "*") {
-			if blobParts.BlobName != "" {
-				return errors.New("cannot combine a wildcarded container name and blob path")
-			}
-
-			blobParts.ContainerName = ""
-		}
-		bUrl = blobParts.URL()
-		jobPartOrder.SourceRoot = bUrl.String()
-	case common.ELocation.File():
-		fromUrl, err := url.Parse(cca.source)
-		if err != nil {
-			return fmt.Errorf("error parsing the source url %s. Failed with error %s", fromUrl.String(), err.Error())
-		}
-		fileParts := azfile.NewFileURLParts(*fromUrl)
-		cca.sourceSAS = fileParts.SAS.Encode()
-		if cca.sourceSAS == "" {
-			return fmt.Errorf("azure files only supports SAS token authentication")
-		}
-		jobPartOrder.SourceSAS = cca.sourceSAS
-		fileParts.SAS = azfile.SASQueryParameters{}
-		fUrl := fileParts.URL()
-		cca.source = fUrl.String()
-
-		// set the clean source root
```

Comment:
Same question as above, re what is doing this check now?, and the stripping of share names with *